pkg/logs/sender/sender_test.go

```diff
@@ -17,10 +19,34 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/logs/message"
 )
 
+type mockStrategy struct {
+}
+
+func (m *mockStrategy) Send(inputChan chan *message.Message, outputChan chan *message.Message, send func([]byte) error) {
+	for msg := range inputChan {
+		outputChan <- msg
+	}
+}
+
+func (m *mockStrategy) Flush(ctx context.Context) {}
+
 func newMessage(content []byte, source *config.LogSource, status string) *message.Message {
 	return message.NewMessageWithSource(content, status, source, 0)
 }
 
+func assertNoMessage(t *testing.T, channel chan *message.Message) {
+	timer := time.NewTimer(10 * time.Millisecond)
```

Comment:
nit: I think we're trying to move to non-time based tests using [`github.com/benbjohnson/clock`](https://github.com/DataDog/datadog-agent/blob/main/pkg/util/sliding_window_test.go#L15)

Reply:
Still a WIP on the tests. But since you are bringing it up, is there a better way to assert something is NOT in a channel? Especially since the failure mode would be a message being written to the output from another thread which could take time.

Reply:
If you're just waiting on a secondary thread to process the data asap, then you can probably ignore my original message.

Reply:
I found a better way!